libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -295,14 +302,68 @@ int32_t AP_BattMonitor::pack_capacity_mah(uint8_t instance) const
     }
 }
 
- /// exhausted - returns true if the voltage remains below the low_voltage for 10 seconds or remaining capacity falls below min_capacity_mah
-bool AP_BattMonitor::exhausted(uint8_t instance, float low_voltage, float min_capacity_mah)
+void AP_BattMonitor::check_failsafes(void)
+{
+    if (hal.util->get_soft_armed()) {
+        for (uint8_t i = 0; i < _num_instances; i++) {
+            const BatteryFailsafe type = check_failsafe(i);
+            if (type <= state[i].failsafe) {
+                continue;
+            }
+
+            int8_t action = 0;
+            const char *type_str = nullptr;
+            switch (type) {
+                case AP_BattMonitor::BatteryFailsafe_None:
+                    return; // should not have been called in this case
+                case AP_BattMonitor::BatteryFailsafe_Low:
+                    action = _params[i]._failsafe_low_action;
+                    type_str = "low";
+                    break;
+                case AP_BattMonitor::BatteryFailsafe_Critical:
+                    action = _params[i]._failsafe_critical_action;
+                    type_str = "critical";
+                    break;
+            }
+
+            gcs().send_text(MAV_SEVERITY_WARNING, "Battery %d %s %.2fV used %.0f mAh", i + 1, type_str,
+                            (double)voltage(i), (double)consumed_mah(i));
+            _has_triggered_failsafe = true;
+            AP_Notify::flags.failsafe_battery = true;
+            state[i].failsafe = type;
+
```

Comment:
Should we/do we need to assert that the critical action setting have >= priority than the low action setting?

Reply:
I think it's a good/worthy idea, but I think it's something that belongs in an arming check, and I'd like to address that in a different/later PR rather then keep creeping the scope on this one. (In general I want to tweak the battery arming checks a bit, so I'd like to just do that in one pass)